lib/Overlay/Overlay.js

```diff
@@ -0,0 +1,591 @@
+*
+* @mixin moon.OverlayContainerSupport
+* @module moonstone/Overlay
+* @public
+*/
+module.exports.Container = {
+
+	/**
+	* @private
+	*/
+	name: 'moon.OverlayContainerSupport',
+
+	/**
+	* Name of the contained control with @{link moon.OverlaySupport}
+	*
+	* @name overlayTarget
+	* @type {String}
+	* @default undefined
+	* @public
+	*/
+
+	/**
+	* List of overlay properties to forward to this container's
+	* {@linke moon.OverlayContainerSupport#overlayTarget}
+	*
+	* @private
+	*/
+	overlayPropertyBindings: [
+		'overlayComponents',
+		'overlayShowing',
+		'overlayClasses',
+		'overlayIcon',
+		'overlayTitle',
+		'overlaySubtitle',
+		'overlayPosition',
+		'overlayAlign',
+		'overlayTransparent'
+	],
+
+	/**
+	* @private
+	*/
+	create: kind.inherit(function (sup) {
+		return function () {
+			sup.apply(this, arguments);
+
+			var fn = this.bindSafely(forwardPropertyChange);
+			this.overlayPropertyBindings.forEach(function (p) {
+				this.observe(p, fn);
+				if (utils.exists(this[p])) fn(null, this[p], p);
+			}, this);
+		}
+	})
+};
+
+/**
+* Sets the default configuration and components for selection overlay. Requires that the control to
+* which it is applied also applies {@link moon.OverlaySupport} or contains a control that does and
+* applies {@link moon.OverlayContainerSupport}.
+*
+* SelectionOverlaySupport adds the styling for a centered icon that can be shown when the `selected`
+* class is applied to the overlaid control.
+*
+* ```
+* {kind: Image, src: 'assets/movie.png', mixins: [Overlay.Support, Overlay.Selection],
+* 	overlayShowing: 'spotlight', overlayComponents: [
+*		{kind: Icon, icon: 'check'}
+* 	]
+* }
+* ```
+*
+* @mixin moon.SelectionOverlaySupport
+* @module moonstone/Overlay
+* @public
+*/
+module.exports.Selection = {
+
+	/**
+	* @private
+	*/
+	name: 'moon.SelectionOverlaySupport',
+
+	/**
+	* @private
+	*/
+	overlayClasses: 'moon-overlay-selection',
+
+	/**
+	* @private
+	*/
+	overlayTransparent: true,
+
+	/**
+	* @private
+	*/
+	overlayPosition: 'centered',
+
+	/**
+	* Icon to be displayed when selected. May either be a URL to an image or the valid name of an
+	* {@link moon.Icon#icon}.
+	*
+	* @name overlayIcon
+	* @type {String}
+	* @default undefined
+	* @public
+	*/
+
+	/**
+	* @private
+	*/
+	bindings: [
+		// to reduce API surface area, we'll only expose overlayIcon and map that to both src and
+		// icon ensuring that only 1 is valued based on the presence of a `.` to indicate a URL
+		{from: 'overlayIcon', to: '$.overlayIcon.src', transform: function (icon) {
+			return (!icon || icon.indexOf('.') >= 0) ? null : icon;
+		}},
+		{from: 'overlayIcon', to: '$.overlayIcon.icon', transform: function (icon) {
+			return (!icon || icon.indexOf('.') == -1) ? null : icon;
+		}}
+	],
+
+	/**
+	* @private
+	*/
+	create: kind.inherit(function (sup) {
+		return function () {
+			this.overlayComponents = this.overlayComponents || [
+				{name: 'overlayIcon', kind: Icon, small: false, icon: 'check', owner: this}
```

Comment:
You might confirm you aren't pulling in the new moonstone font file which changed the code point for 'check' (e.g. you installed it locally and that's what it's using). I'll check things out but I believe I noticed the same behavior and that was the cause since I was reviewing the ticket in the same window.